### PR TITLE
Lb/better logs

### PR DESCRIPTION
**Pull request by mynameiswillporter**
This change decreases logging noise as well as converts our application logs entirely to JSON.  This is to simplify programmatic parsing of the logs for integration with our elastic stack.

### CODE REVIEW

File: infrastructure/provision-rails-assets.yml [hunk 1]
```diff
@@ -290,6 +313,32 @@
...
+  tasks:
+    - name: create downloads directory
+      file:
+        path: "/home/{{ app_name }}/downloads"
+        state: directory
+        owner: "{{ app_name }}"
+        group: "users"
+        mode: "u=rwx,g=rwx,o=rx"
+    - name: download filebeat
+      get_url:
```
Comment (hut8) [inline infrastructure/provision-rails-assets.yml]: Let's get this from the elasticsearch repo

File: infrastructure/provision-rails-assets.yml [hunk 2]
```diff
@@ -219,6 +219,29 @@
         insertafter='^http {'
         line='        charset utf-8;'
       notify: restart nginx
+    - name: "add json logging format to nginx"
+      blockinfile:
+      dest: /etc/nginx/nginx.conf
+      insertafter: '^http {'
+      content: |
+        log_format logstash_json '{ "@timestamp": "$time_iso8601", '
```
Comment (hut8) [inline infrastructure/provision-rails-assets.yml]: Neat. I had no idea about this. What about json escaping, though? Like if someone puts a `"` in their useragent string?

File: config/initializers/colorful_logger.rb [hunk 3]
```diff
@@ -21,6 +21,19 @@ def call(severity, _time, _progname, msg)
     formatted_severity = sprintf('%-3s', SEVERITY_TO_TAG_MAP[severity])
     color = SEVERITY_TO_COLOR_MAP[severity]
```
Comment (hut8) [inline config/initializers/colorful_logger.rb]: This whole file is stupid; we should get rid of it

File: infrastructure/provision-rails-assets.yml [hunk 4]
```diff
@@ -219,6 +219,30 @@
...
+          log_format logstash_json '{ "@timestamp": "$time_iso8601", '
+                       '"@fields": { '
+                       '"remote_addr": "$remote_addr", '
+                       '"remote_user": "$remote_user", '
+                       '"body_bytes_sent": "$body_bytes_sent", '
+                       '"request_time": "$request_time", '
+                       '"status": "$status", '
+                       '"request": "$request", '
+                       '"request_method": "$request_method", '
+                       '"http_referrer": "$http_referer", '
```
Comment (hut8) [inline infrastructure/provision-rails-assets.yml:244]: What happens when someone has a `"` in their useragent string?

File: infrastructure/provision-rails-assets.yml [hunk 5]
```diff
@@ -290,6 +314,32 @@
         name: "nginx"
         state: "restarted"
 
+- name: centralize logging
+  hosts: web
+  user: root
+  tasks:
+    - name: create downloads directory
```
Comment (hut8) [inline infrastructure/provision-rails-assets.yml]: We don't need this if we pull it from the repo

File: infrastructure/provision-rails-assets.yml [hunk 6]
```diff
@@ -290,6 +314,32 @@
...
+    - name: install filebeat
+      apt:
+        deb: "/home/{{ app_name }}/downloads/filebeat-5.3.2-amd64.deb"
+    - name: configure filebeat
+      copy:
+        src: "files/filebeat.yml"
+        dest: "/etc/filebeat/filebeat.yml"
+        mode: "0644"
+    - name: start filebeat
+      command: "/etc/init.d/filebeat start"
```
Comment (hut8) [inline infrastructure/provision-rails-assets.yml]: We should use 
```
service: 
  name: filebeat
  state: started
```

**Review by hut8 — CHANGES_REQUESTED** [review]

File: config/initializers/colorful_logger.rb (hunk 3 above)
Comment (mynameiswillporter) [reply config/initializers/colorful_logger.rb]: I originally erased the whole file, but many logs were still not in JSON format.  I removed the coloring and custom log levels from the file and renamed it JSON logger.  This allows it to put the rest of the logs in JSON format.

File: infrastructure/provision-rails-assets.yml (hunk 4 above)
Comment (mynameiswillporter) [reply infrastructure/provision-rails-assets.yml:244]: Good call, I updated the escaping from 'default' to 'json'.

**Review by hut8 — APPROVED** [review]

File: infrastructure/provision-rails-assets.yml (hunk 4 above)
Comment (hut8) [reply infrastructure/provision-rails-assets.yml:244]: I just deployed this and realized:

```
The escape parameter (1.11.8) allows setting json or default characters escaping in variables, by default, default escaping is used.
```

We're on nginx 1.10.2 ☹️

**Pull request merged**